crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -365,60 +371,163 @@ where
 
         let topics = filter.has_topics().then(|| filter_params.flat_topics.clone());
 
-        // derive bloom filters from filter input
-        let address_filter = FilteredParams::address_filter(&filter.address);
-        let topics_filter = FilteredParams::topics_filter(&topics);
+        // Create log index filter
+        let mut log_index_filter =
+            LogIndexFilter::new(from_block..=to_block, self.max_headers_range);
+        if let Some(filter_address) = filter.address.as_ref() {
+            log_index_filter.install_address_filter(&self.provider, filter_address)?;
+        }
+        if let Some(topics) = topics.as_ref() {
+            log_index_filter.install_topic_filter(&self.provider, topics)?;
```

Comment:
should some fo this stuff be moved to the provider? so that e.g. reth-indexer type code doesn't need to reimplement parts of it and can re-use it?